src/shogun/features/DotFeatures.cpp

```diff
@@ -200,6 +200,44 @@ SGVector<float64_t> CDotFeatures::get_mean()
 	return mean;
 }
 
+SGVector<float64_t> CDotFeatures::get_std(bool colwise) const
+{
+	auto num=get_num_vectors();
+	auto dim=get_dim_feature_space();
+	ASSERT(num>0)
+	ASSERT(dim>0)
+
+	auto mean = get_mean();
+	if (!colwise)
+	{
+		auto global_mean = linalg::sum(mean) / dim;
+		linalg::set_const(mean, global_mean);
+	}
+
+	linalg::scale(mean, mean, -1.0);
+	SGVector<float64_t> std(dim);
+	for (index_t i = 0; i < num; ++i)
+	{
+		auto m = mean.clone();
```

Comment:
I think there might be a bug here: when the columnwise std-dev is computed, the column-wise mean would have to be substracted instead of the global one. Or am I confused?

Reply:
yeah.... and in case if colwise = true mean is the colwise mean

Reply:
note i use the same container (mean) to get the std-dev.... avoid code redundancy and since i need to use add_to_dense_vector....

Reply:
see https://github.com/shogun-toolbox/shogun/pull/4548/files#diff-e3b574b8a2ef609bae51ca0e7298680cR214

Reply:
nice! :)